presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/QueryBuilder.java

```diff
@@ -62,31 +88,56 @@ public String buildSql(String catalog, String schema, String table, List<JdbcCol
         }
         sql.append(quote(table));
 
-        List<String> clauses = toConjuncts(columns, tupleDomain);
+        List<TypeAndValue> accumulator = new ArrayList<>();
+
+        List<String> clauses = toConjuncts(columns, tupleDomain, accumulator);
         if (!clauses.isEmpty()) {
             sql.append(" WHERE ")
                     .append(Joiner.on(" AND ").join(clauses));
         }
 
-        return sql.toString();
+        PreparedStatement statement = connection.prepareStatement(sql.toString());
+
+        for (int i = 0; i < accumulator.size(); i++) {
+            TypeAndValue typeAndValue = accumulator.get(i);
+            if (typeAndValue.getType().equals(BigintType.BIGINT)) {
+                statement.setLong(i + 1, (Long) typeAndValue.getValue());
+            }
+            else if (typeAndValue.getType().equals(DoubleType.DOUBLE)) {
+                statement.setDouble(i + 1, (Double) typeAndValue.getValue());
+            }
+            else if (typeAndValue.getType().equals(BooleanType.BOOLEAN)) {
+                statement.setBoolean(i + 1, (Boolean) typeAndValue.getValue());
+            }
+            else {
+                throw new UnsupportedOperationException("Can't handle type: " + typeAndValue.getType().getClass().getName());
+            }
+        }
+
+        return statement;
+    }
+
+    private static boolean isAcceptedType(Type type)
+    {
+        return type != null && (type.equals(BigintType.BIGINT) || type.equals(DoubleType.DOUBLE) || type.equals(BooleanType.BOOLEAN));
```

Comment:
type should never be null, so we shouldn't check for it this way. this implementation implies that it can validly be null